### DISCUSSION TITLE
Stuck due to Android soft keyboard

### DISCUSSION THREAD

**Discussion by tsctsc6** [AvaloniaUI/Avalonia#15987] [Q&A] [answered]
### Describe the bug

I created a project that was written on the [document](https://docs.avaloniaui.net/docs/get-started/test-drive/). In some cases, the app will get stuck.

### To Reproduce

* Input a number on the textbox.
* Collapse the soft keyboard firstly and then click the button, the app run normally.
* Input a number on the textbox again.
* Don't collapse the soft keyboard and click the button directly, the app would be stuck.

### Expected behavior

Don't collapse the soft keyboard and click the button directly, the soft keyboard would be collapsed and the app would run normally.

### Avalonia version

11.0.2

### OS

Android

### Additional context

* This bug can be reproduced on other devices.
* The app run normally on Android Emulator, I think it's due to Android Emulator without a soft keyboard.
* Don't collapse the soft keyboard and click other controls which "IsReadonly=True" directly, the app would be stuck too.

**Comment by timunie** [comment]
Can you please try a newer Avalonia version? At minimum 11.0.10 or even better 11.1 beta. Thx 🙏

**Comment by tsctsc6** [comment]
I was wondering why I was told to get .net 7. The template default is .net7 and Avalonia version is 11.0.2.

**Comment by tsctsc6** [comment]
No problems with the new version(11.1 beta), Thx.

**Comment by tsctsc6** [comment]
So could you please update the Avalonia.template in VS?

**Comment by timunie** [comment]
@tsctsc6 they are updated but you need to pull them on your own from time to time. We cannot do this as we don't own VS. You can simply run the install command again to do so. https://docs.avaloniaui.net/docs/get-started/install

**Comment by tsctsc6** [comment]
But I just installed Avalonia yesterday!

**Comment by tsctsc6** [comment]
I run ```dotnet new install Avalonia.Templates``` just now,
and the result show that Avalonia.Templates version is 11.0.10.1.
Then I create new a project, still .net 7 and 11.0.2.

**Reply by timunie** [reply]
Never seen this tbh. Maybe you have installed them twice by accident?

**Reply by thevortexcloud** [reply]
Try running `dotnet new update`. The whole process for always using the latest templates in dotnet is fairly obtuse. It really should check for updates whenever you try to use it (after all, it's a new project) but it does not. I often accidentally use old templates.

**Reply by tsctsc6** [reply]
I try running ```dotnet new update```, the result is "All template packages are up to date."

**Reply by thevortexcloud** [reply, marked as answer]
Does it work if you run `dotnet new avalonia.app`? 

It's also possible you actually have two copies of .NET SDK installed. In which case you may be updating the wrong SDK.

**Reply by tsctsc6** [reply]
After run ```dotnet new avalonia.app```, it does .net 8 and 11.0.10. But it only is a window app project, without android project, etc.
The way I created the project before was this below:
![image](https://github.com/AvaloniaUI/Avalonia/assets/93368014/98a84ece-53f9-4bdb-8bc2-c01793358609)

**Reply by thevortexcloud** [reply]
Ah, that's the Visual Studio extension. I am not entirely sure how that works. You may need to update it?

>  without window app project, android project, etc.

The template for that is `avalonia.xplat` but I don't think it lets you pick a specific platform, it just adds all of them. But you can easily manually delete the ones you don't want.

**Reply by tsctsc6** [reply]
> So could you please update the Avalonia.template in VS?

So...I mean are you considering updating the Visual Studio extension?

**Reply by tsctsc6** [reply]
Anyway, problem solved. Thanks.

**Reply by thevortexcloud** [reply]
The Avalonia templates and the extension are different things. The extension adds the wizard for setting up the projects. The templates are what the SDK itself uses to make the projects. I am not clear if/how the extension uses the templates or if it has its own. On Rider at least it always uses the SDK templates as the wizard does not exist for anything but Visual Studio.

**Reply by tsctsc6** [reply]
Oh, someone already mentioned this a few months ago.[https://github.com/AvaloniaUI/TemplateStudio/issues/32#issue-2263553075](https://github.com/AvaloniaUI/TemplateStudio/issues/32#issue-2263553075)
